Zend/zend_smart_str.c

```diff
@@ -169,3 +169,30 @@ ZEND_API void ZEND_FASTCALL _smart_string_alloc(smart_string *str, size_t len)
 		str->c = erealloc2(str->c, str->a + 1, str->len);
 	}
 }
+
+ZEND_API void ZEND_FASTCALL smart_str_append_truncated(smart_str *str, zend_string *value, size_t length)
+{
+	smart_str_append_escaped(str, ZSTR_VAL(value), MIN(length, ZSTR_LEN(value)));
+
+	if (ZSTR_LEN(value) > length) {
+		smart_str_appendl(str, "...", sizeof("...")-1);
+	}
+}
+
+ZEND_API void ZEND_FASTCALL smart_str_append_scalar(smart_str *dest, zval *value) {
```

Comment:
Should this accept truncation length as parameter, or using `exception_string_param_max_len` is okay ?

Reply:
I would have it as parameter - exception length is not quite the same secnario as in exceptions you will have multiple args, but in a trivial standalone case no.